### PR TITLE
sqltypes: improve Result.Copy

### PR DESCRIPTION
**Pull request by sougou**
Consolidate allocations to one per row, since we know that uses
less CPU.
Also renamed the rest of the protos.

### CODE REVIEW

File: go/sqltypes/value.go [hunk 1]
```diff
@@ -46,11 +46,10 @@ type Value struct {
...
 // MakeTrusted makes a new Value based on the type.
-// If val is nil, the Value is NULL. If the value is an integral,
-// then val must be in its cannonical form. This function should
-// only be used if you know the value and type conform to the rules.
-// Every place this function is called, a comment is needed that explains
-// why it's justified.
+// If the value is an integral, then val must be in its cannonical
+// form. This function should only be used if you know the value
+// and type conform to the rules.  Every place this function is
+// called, a comment is needed that explains why it's justified.
```
Comment (enisoc) [inline go/sqltypes/value.go]: You didn't follow this rule in `Result.Copy()` :P

**Comment by enisoc** [comment]
LGTM

File: go/sqltypes/value.go (hunk 1 above)
Comment (sougou) [inline go/sqltypes/value.go]: Actually, the rule applies only outside sqltypes :). You'll see that I've used that function all over the place inside the package.
But fair is fair, I'll change the comment to clarify this exception.

**Comment by sougou** [comment]
I accidentally pushed my proto rename change into this PR. Could someone review it?

**Comment by enisoc** [comment]
LGTM

**Pull request merged**